torch_ttnn/backend.py

```diff
@@ -122,6 +134,50 @@ def aten_backend(
     gm.graph.lint()
     gm.recompile()
 
+    # Get the memory manager object for memory analysis
+    if option.run_mem_analysis:
+        for p in passes:
+            if isinstance(p, MemoryPass):
+                option._memory_manager = p.memory_manager
+
+    # Run eviction opt pass if enabled
+    if option.run_eviction_opt == True:
+        assert (
+            option.run_mem_analysis == True
+        ), "Eviction pass depends on memory analysis pass!"
+        from torch_ttnn.passes.eviction_pass import EvictionPass
+
+        mm = option._memory_manager
+        nth_eviction = 1
+        # See if SRAM overflows and eviction is required
+        while mem_utils.check_sram_overflow(mm) is True:
+            # if check_sram_overflow(mm) is True:
+            guilty_op, tensors_to_evict = mem_utils.which_tensors_to_evict(mm)
+            # This indicates splitting is required
+            if tensors_to_evict == -1:
+                break
+            # Run a eviction pass to remove tensors from device
+            # This pass only evicts tensors for a single ttnn op which overflows the SRAM
+            # Multiple overflows require multiple run of this pass
+
+            passes = [
+                EvictionPass(mm, guilty_op, tensors_to_evict),
+                GraphvizPass(f"eviction-pass-{nth_eviction}"),
+                MemoryPass(),
+            ]
+            pm = PassManager(passes=passes)
+            gm, modified = pm(gm)
+
+            gm.graph.lint()
+            gm.recompile()
```

Comment:
@kevinwuTT @mcw-zwakeelTT guys, are we supposed to call lint/recompile everytime? Whats the cost on the compile time? Does this run all passes again?

Reply:
lint() & recompile() is required to be called whenever FX graph is modified. This ensures the modified graph is valid. I don't think recompile() goes through our compilation pipeline, it regenerates the forward() function to reflect the modified graph changes.

Reply:
From my understanding, when you manually modify the `gm.graph`, you're actually modifying a separate copy of the graph, `GraphModule._graph` [link](https://github.com/pytorch/pytorch/blob/bedf96d7ffe74b34bcfe52c7ae1ae05f40d6c8ee/torch/fx/graph_module.py#L503-L507) So you will need to call `gm.recompile()` to update the actual `graph` member if you want to access gm.graph at a later time. The [recompile](https://github.com/pytorch/pytorch/blob/bedf96d7ffe74b34bcfe52c7ae1ae05f40d6c8ee/torch/fx/graph_module.py#L755-L788) function seems to be doing mostly reassignment. I can't tell immediately what the cost of this call is, but we could definitely optimize the amount of times we run recompile. I can add this as a task to look at a later time?